standards/SPIFFE_Workload_API.md

```diff
@@ -360,6 +356,129 @@ When interfacing with legacy JWT validators, the `FetchJWTBundles` method can be
+
+```protobuf
+service SpiffeWorkloadAPI {
+  /////////////////////////////////////////////////////////////////////////
+  // WIT-SVID Profile
+  /////////////////////////////////////////////////////////////////////////
+
+  // Fetch WIT-SVIDs for all SPIFFE identities the workload is entitled to. If
+  // the identities which the workload is entitled to change, or, they are 
+  // renewed, the server will stream subsequent messages to the client.
+  //
+  // Must return Unimplemented where Workload API endpoint does not support the
+  // WIT-SVID profile.
+  rpc FetchWITSVID(WITSVIDRequest) returns (stream WITSVIDResponse);
+
+  // Fetch trust bundles. Useful for clients that only need to validate SVIDs
+  // without obtaining an SVID for themselves. As this information changes,
+  // subsequent messages will be streamed from the server.
+  //
+  // Must return Unimplemented where Workload API endpoint does not support the
+  // WIT-SVID profile.
+  rpc FetchWITBundles(WITBundlesRequest) returns (stream WITBundlesResponse);
+
+  // ... RPCS for other profiles ...
+}
+
+// The WITSVIDRequest message conveys parameters for requesting a WIT-SVID.
+// There are currently no such parameters.
+message WITSVIDRequest {}
+
+// The WITSVIDResponse message carries WIT-SVIDs .
+message WITSVIDResponse {
+  // Required.
+  // A list of WITSVID messages, each of which includes a single WIT-SVID and
+  // its private key.
+  repeated WITSVID svids = 1;
+}
+
+// The WITSVID message carries a single SVID and all associated information,
+// including the WIT bundle for the trust domain.
+message WITSVID {
+  // Required.
+  // The SPIFFE ID of the SVID in this entry
+  string spiffe_id = 1;
+
+  // Required.
+  // Encoded WIT SVID using compact JWS serialization.
+  string wit_svid = 2;
+
+  // Required.
+  // JWK (RFC 7517) encoded private key for the WIT SVID.
+  string wit_svid_key = 3;
+
+  // Optional.
+  // An operator-specified string used to provide guidance on how this identity
+  // should be used by a workload when more than one SVID is returned.
+  // For example, `internal` and `external` to indicate a SVID for internal or
+  // external use, respectively.
+  string hint = 4;
+}
+
+// The WITBundlesRequest message conveys parameters for requesting WIT
+// bundles. There are currently no such parameters.
+message WITBundlesRequest {}
+
+// The WITBundlesResponse message carries a map of trust bundles the workload
+// should trust.
+message WITBundlesResponse {
+  // Required.
+  // WIT trust bundles belonging to trust domains that the workload should
+  // trust, keyed by the SPIFFE ID of the trust domain.
+  // Bundles are encoded in JWK set (RFC 7517) format.
+  map<string, string> bundles = 1;
+}
+```
+
+### 7.2 Profile RPCs
+
+#### 7.2.1 FetchWITSVID
+
+The `FetchWITSVID` RPC streams back WIT-SVIDs. 
+
+The `WITSVIDRequest` request message is currently empty and is a placeholder for future expansion.
```

Comment:
I know we discussed this before but I don’t remember, what was the reason we don’t support requesting a specific SPIFFE-ID here?

Reply:
@strideynet Isn't the whole point of a `hint` field that the sender cannot assume the receiver understands/respects it? Or are you saying that a server that does not recognize this field will return an error?

Reply:
Let's discuss later today in SIG-SPEC. My feeling is that from a security perspective it's better to issue/release only a single credential to a workload in situations where it knows what SPIFFE-ID it requires. Issuing all and the workload just picking one seems unnecessary and when avoidable better for security.

Reply:
We discussed this during SIG-Spec, and came to the consensus that it would make sense to include this field for the WIT-SVID profile. I will adjust the spec accordingly.